iothub/device/src/Authentication/IotHubConnectionInfo.cs

```diff
@@ -178,14 +177,63 @@ async Task<string> IAuthorizationProvider.GetPasswordAsync()
             }
         }
 
-        public Uri BuildLinkAddress(string path)
+        public override bool Equals(object obj)
```

Comment:
Is this method even needed? I found one on another object that had no references, so I assumed it was created with no applicable reason in mind.

Reply:
I believe this particular one is being referenced in our connection pooling logic to resolve the identity of the AmqpUnit to be removed from the pool. I can add a comment calling that out.